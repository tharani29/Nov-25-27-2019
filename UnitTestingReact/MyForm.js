import React, { Component } from 'react';

class MyForm extends Component {
	constructor(props) {
		super(props);
		this.state = { name: "", message: ""}	
	}	
	
	textChanged(e) {
		this.setState({ name: e.target.value.trim() });
	}
	
	buttonClicked() {
		this.setState({
			message: `Hi ${this.state.name}`
		})
	}
	
  render() {
    return (<div>
		<input type="text" onChange={this.textChanged.bind(this)} placeholder="Name"/>
		<br/>
    	<button onClick={this.buttonClicked.bind(this)}>Click</button>
		<h3>{this.state.message}</h3>
	</div>);
  }
}

export default MyForm;
