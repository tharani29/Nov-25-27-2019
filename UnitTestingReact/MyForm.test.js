import React from 'react';
import { mount } from 'enzyme';
import MyForm from './MyForm';

describe("MyForm", () => {
	
	it("should call the textChanged function", () => {
		const component = mount(<MyForm/>);
		let textbox = component.find("input");
		textbox.simulate('change', { target: { value: "Sam"} });
		expect(component).toMatchSnapshot();
		expect(component.state('name')).toEqual("Sam");
		component.unmount();
	});
	
	it("should call the buttonClick function", () => {
		const component = mount(<MyForm/>);
		let textbox = component.find("input");
		textbox.simulate('change', { target: { value: "Sam"} });
		let button = component.find("button");
		button.simulate('click');
		expect(component).toMatchSnapshot();
		expect(component.state('message')).toEqual("Hi Sam");
		component.unmount();
	});
	
});