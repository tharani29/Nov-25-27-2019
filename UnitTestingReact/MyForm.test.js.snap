// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyForm should call the buttonClick function 1`] = `
<MyForm>
  <div>
    <input
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <br />
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h3>
      Hi Sam
    </h3>
  </div>
</MyForm>
`;

exports[`MyForm should call the textChanged function 1`] = `
<MyForm>
  <div>
    <input
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
    <br />
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h3 />
  </div>
</MyForm>
`;
